include/dkm_parallel.hpp

```diff
@@ -115,18 +118,20 @@ Returns a std::tuple containing:
 Implementation details:
 This implementation of k-means uses [Lloyd's Algorithm](https://en.wikipedia.org/wiki/Lloyd%27s_algorithm)
 with the [kmeans++](https://en.wikipedia.org/wiki/K-means%2B%2B)
-used for initializing the means.
+used for initializing the means. An optional argument for a seeded generator is also provided.
 */
 template <typename T, size_t N>
 std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> kmeans_lloyd_parallel(
-	const std::vector<std::array<T, N>>& data, const clustering_parameters<T>& parameters) {
+	const std::vector<std::array<T, N>>& data,
+	const clustering_parameters<T>& parameters, 
+	CustomGenerator& rand_engine=CustomGenerator(0)) 
```

Comment:
There is an issue with this default argument here:
- since the `rand_engine` is passed by-reference, it cannot have a default value unless we `const` it. If we `const` it, then we have to `const_cast` it before passing it to the `details::random_plusplus_parallel()` function.
- another solution would be to not pass the argument's value by-reference
- there is also [this](https://stackoverflow.com/questions/1059630/default-value-to-a-parameter-while-passing-by-reference-in-c) solution for creating a static instance of the `CustomGenerator`, but I'm not sure what's the base way to go here. @friniaccusonus any thoughts on this?